### PR TITLE
Fix analyzer warnings in parser_test.dart

### PR DESCRIPTION
**Pull request by davidmorgan**

### CODE REVIEW

File: test/parser_test.dart [hunk 1]
```diff
@@ -329,14 +329,14 @@ main() {
       MAP_WITH_SOURCE_LOCATION,
       MAP_WITH_SOURCE_LOCATION_AND_NAME
     ]) {
-      var mapping = parseJson(expected);
+      var mapping = parseJson(expected) as SingleMapping;
```
Comment (jacob314) [inline test/parser_test.dart]: nit: why not
SingleMapping mapping = parseJson(expected);
?
Then the second "as SingleMapping" can be removed.

File: test/parser_test.dart [hunk 2]
```diff
@@ -329,14 +329,14 @@ main() {
...
       expect(mapping.toJson(), equals(expected));
 
-      mapping = parseJsonExtended(expected);
+      mapping = parseJsonExtended(expected) as SingleMapping;
       expect(mapping.toJson(), equals(expected));
     }
     // Invalid for this case
-    expect(() => parseJson(SOURCE_MAP_BUNDLE as dynamic), throws);
+    expect(() => parseJson(SOURCE_MAP_BUNDLE as dynamic), throws) as MappingBundle;
```
Comment (jacob314) [inline test/parser_test.dart]: this line makes no sense. Does this even compile?

**Review by jacob314 — APPROVED** [review]
lgtm once the strange as SingleMapping in the expect statement is removed.

File: test/parser_test.dart (hunk 1 above)
Comment (davidmorgan) [reply test/parser_test.dart]: Done.

File: test/parser_test.dart (hunk 2 above)
Comment (davidmorgan) [reply test/parser_test.dart]: Sorry, copy+paste error. Should have added to the next line...

**Review by jacob314 — APPROVED** [review]

**Pull request merged**